### PR TITLE
harness-generate refactoring, documentation review

### PR DESCRIPTION
**Pull request by ddelpiano**
Closes [CH-X](https://metacell.atlassian.net/browse/CH-X)

# Implemented solution

small refactoring for harness-generate to improve some functionalities and tackle some corner cases not considered before.

# How to test this PR

Install the cli tools and run harness-generate following the documentation of the helper.
Different modes and parameters will generate different output and stubs.

# Sanity checks:
- [x] The pull request is explicitly linked to the relevant issue(s)
- [ ] The issue is well described: clearly states the problem and the general proposed solution(s)
- [x] In this PR it is explicitly stated how to test the current change
- [x] The labels in the issue set the scope and the type of issue (bug, feature, etc.)
- [x] The relevant components are indicated in the issue (if any)
- [x] All the automated test checks are passing
- [x] All the linked issues are included in one Sprint
- [x] All the linked issues are in the Review state
- [x] All the linked issues are assigned

# Breaking changes (select one):
- [x] The present changes do not change the preexisting api in any way
- [ ] This PR and the issue are tagged as a `breaking-change` and the migration procedure is well described [above](#implemented-solution)

# Possible deployment updates issues (select one):
- [x] There is no reason why deployments based on CloudHarness may break after the current update
- [ ] This PR and the issue are tagged as `alert:deployment`

### Test coverage (select one):
- [ ] Tests for the relevant cases are included in this pr
- [x] The changes included in this pr are out of the current test coverage scope

### Documentation (select one):
- [x] The documentation has been updated to match the current changes
- [ ] The changes included in this PR are out of the current documentation scope

### Nice to have (if relevant):
- [ ] Screenshots of the changes
- [ ] Explanatory video/animated gif

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,30 +68,34 @@ CloudHarness provides the following command line tools to help application scaff
...
-Python 3.9 must be installed.
+
+Python 3.10 must be installed.
 
 It is recommended to setup a virtual environment.
 With conda:
+
 ```bash
-conda create --name ch python=3.12
+conda create --name ch python=3.10
```
Comment (filippomc) [inline README.md:93]: why python 3.10 and not 3.12?

File: README.md [hunk 2]
```diff
@@ -120,26 +130,183 @@ A JRE is needed to run the code generators based on openapi-generator.
...
 
 ### Generate server and client code from openapi
-To (re)generate the code for your applications, run `harness-generate` from the root.
-The script will look for all openapi applications, and regenerate the Flask server code and documentation.
-Note: the script will eventually override any manually modified file. To avoid that, define a file openapi-generator-ignore.
+
+To (re)generate the code for your applications, run `harness-generate`.
+`harness-generate` is a command-line tool used to generate client code, server stubs, and model libraries for applications. It walks through the filesystem inside the `./applications` folder to create and update application scaffolding. The tool supports different generation modes and allows for both interactive and non-interactive usage.
+
+#### Usage
```
Comment (filippomc) [inline README.md:211]: It looks a lot of details for the main readme, what about moving to docs? There is also this one: https://github.com/MetaCell/cloud-harness/blob/2d03fd84b0532274c05e1aee104edf32ecb91e94/docs/applications/harness-application.md. Maybe we can create docs/code-generation.md and merge this section and that file?

**Review by filippomc — APPROVED** [review]
Looks good
